tools/setup/generate_zulip_bots_static_files.py

```diff
@@ -55,5 +56,19 @@ def create_png_from_svg(svg_path: str, destination_dir: Optional[str]=None) -> s
     cairosvg.svg2png(url=svg_path, write_to=png_path)
     return png_path
 
+def create_square_image(png_path: str) -> str:
+    img = Image.open(png_path)
+    if img.height == img.width:
+        return png_path
+
+    size = max(img.height, img.width)
+    new_img = Image.new("RGBA", (size, size), color=None)
+    padding = int(abs(img.height - img.width) / 2)
+    position = (0, padding) if img.height < img.width else (padding, 0)
+    new_img.paste(img, position)
+    new_path = os.path.join(tempfile.gettempdir(), os.path.basename(png_path))
+    new_img.save(new_path)
+    return new_path
```

Comment:
I think this is probably the point at which we should make these more of a real thing.  Otherwise, every time we run this command, it'll make a new image file in a temporary directory, which will eventually add up!

One option would be two functions to the `optimize-svg` stage of things (i.e. something that generates e.g. `github_avatar.png` PNG files; that we can commit to Git; this is only a good idea is the resulting PNGs are stable, which they may not be).  The other option would be to make this generating part of `update-prod-static`; if it's fast, that's a fine answer, but ideally we don't want every `upgrade-zulip-from-git` in production to need to regenerate these.

Some direction background is that we intend to make a "create this bot" flow in the UI that would use these avatars, which is why I'm thinking we want to make this system for generating avatar-usable images from the SVGs robust and usable in production.

Reply:
Thanks for the background, @timabbott. Now, it makes a lot more sense to me why you suggested adding these pngs to the repo, before. I'll play around a little with this, and see what I can come up with.